File: src/main/java/com/libraryapp/domain/user/UserRepository.java
package com.libraryapp.domain.user;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.util.Optional;

public interface UserRepository extends JpaRepository<Users, Long> {

    @Query("SELECT u FROM Users u JOIN FETCH u.loanBookHistories WHERE u.name = :name")
    Optional<Users> findByNameWithLoanHistories(String name);

    Optional<Users> findByName(String name);
}

File: src/main/kotlin/com/libraryapp/domain/user/UserRepository.kt
package com.libraryapp.domain.user

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query

interface UserRepository : JpaRepository<Users, Long>, UserRepositoryCustom {

    @Query("SELECT u FROM Users u JOIN FETCH u.loanBookHistories WHERE u.name = :userName")
    fun findByNameWithLoanHistories(userName: String): Users?

    @Query("SELECT DISTINCT u FROM Users u LEFT JOIN FETCH u.loanBookHistories")
    fun findAllUsersWithLoanHistories(): List<Users>

    fun findByName(userName: String): Users?
}